src/main/java/com/ryandw11/structure/commands/cstruct/TestSpawnCommand.java

```diff
@@ -198,6 +198,9 @@ void canSpawn(Player p, Structure structure, Block block, Chunk chunk) {
         if (!CustomStructures.getInstance().getStructureHandler().validDistance(structure, block.getLocation()))
             quickSendMessage(p, "&cFailed Distance Limitation test! Cannot spawn this close to another structure!");
 
+        if (!CustomStructures.getInstance().getStructureHandler().validSameDistance(structure, block.getLocation()))
+            quickSendMessage(p, "&cFailed Distance Limitation test! Cannot spawn this close to the same structure!");
+
```

Comment:
I'm not that familiar with the codebase, but I'm wondering why this is checked both here and structure.canSpawn(). Couldn't we just call structure.canSpawn() to check all conditions? I understand that would generate problems to give separate error messages, but going forward might make it cleaner.
Then again, sorry if this comment makes no sense as I am really unfamiliar with the codebase

Reply:
I just added the DistanceFromSame option, I don't really want to change any of the code of ryandw11..